pkg/registry/batch/job/strategy.go

```diff
@@ -236,8 +243,12 @@ func generateSelector(obj *batch.Job) {
 	if obj.Spec.Selector.MatchLabels == nil {
 		obj.Spec.Selector.MatchLabels = make(map[string]string)
 	}
-	if _, found := obj.Spec.Selector.MatchLabels["controller-uid"]; !found {
-		obj.Spec.Selector.MatchLabels["controller-uid"] = string(obj.ObjectMeta.UID)
+
+	// TODO: when job labels have both legacy labels and the official Kubernetes ones we should replace this selector to use batch.ControllerUid
+	for _, value := range []string{batch.LegacyControllerUid} {
```

Comment:
We don't need the for.

But, as discussed in the other thread, we can probably just use the new labels already.